src/matmul.jl

```diff
@@ -570,6 +570,86 @@ Base.@constprop :aggressive function _symm_hemm_generic!(C, tA, tB, A, B, alpha,
     _generic_matmatmul!(C, wrap(A, tA), wrap(B, tB), alpha, beta)
 end
 
+Base.@constprop :aggressive function generic_matmatmul_wrapper!(C::StridedMatrix{T}, tA, tB, A::StridedVecOrMat{T}, B::StridedVecOrMat{T},
+                                    α::Number, β::Number, val::BlasFlag.SyrkHerkGemm) where {T<:Number}
+    mA, nA = lapack_size(tA, A)
+    mB, nB = lapack_size(tB, B)
+    if any(iszero, size(A)) || any(iszero, size(B)) || iszero(α)
+        matmul_size_check(size(C), (mA, nA), (mB, nB))
+        return _rmul_or_fill!(C, β)
+    end
+    matmul2x2or3x3_nonzeroalpha!(C, tA, tB, A, B, α, β) && return C
+
+    if A === B
+        tA_uc = uppercase(tA) # potentially strip a WrapperChar
+        aat = (tA_uc == 'N')
+        blasfn = _valtypeparam(val)
+        if blasfn == BlasFlag.SYRK && T <: Union{Real,Complex} && (iszero(β) || issymmetric(C))
+            return copytri!(generic_syrk!(C, A, false, aat, α, β), 'U')
+        elseif blasfn == BlasFlag.HERK && (iszero(β) || ishermitian(C))
+            return copytri!(generic_syrk!(C, A, true, aat, α, β), 'U', true)
+        end
+    end
+
+    return _generic_matmatmul!(C, wrap(A, tA), wrap(B, tB), α, β)
+end
+
+function generic_syrk!(C::StridedMatrix{T}, A::StridedVecOrMat{T}, conjugate::Bool, aat::Bool, α, β) where {T<:Number}
+    require_one_based_indexing(C, A)
+    nC = checksquare(C)
+    m, n = size(A, 1), size(A, 2)
+    mA = aat ? m : n
+    if nC != mA
+        throw(DimensionMismatch(lazy"output matrix has size: $(size(C)), but should have size $((mA, mA))"))
+    end
+
+    _rmul_or_fill!(C, β)
+    @inbounds if !conjugate
+        if aat
+            for k ∈ 1:n, j ∈ 1:m
+                αA_jk = α * A[j, k]
```

Comment:
In the non-commutative case, we agreed to interpret the 5-arg-mul as having alpha (and beta) being multiplied from the right.

```suggestion
                αA_jk = A[j, k] * α
```